src/shims/unix/fs.rs

```diff
@@ -1660,6 +1690,45 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
     }
+
+    fn isatty(&mut self, fd: &OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+        #[cfg(unix)]
+        {
+            let fd = this.read_scalar(fd)?.to_i32()?;
+            if let Some(fd) =
+                this.machine.file_handler.handles.get(&fd).and_then(|fd| fd.as_unix_host_fd())
+            {
+                // "returns 1 if fd is an open file descriptor referring to a terminal;
+                // otherwise 0 is returned, and errno is set to indicate the error"
+                let is_tty = unsafe { libc::isatty(fd) };
+                if is_tty == 0 {
+                    let enotty = this.eval_libc("ENOTTY")?;
+                    let errno = std::io::Error::last_os_error()
+                        .raw_os_error()
+                        .map_or(enotty, Scalar::from_i32);
+                    this.set_last_error(errno)?;
+                }
+                Ok(is_tty)
+            } else {
+                // FIXME: We are attemping to use a Unix interface on a non-Unix platform. It
```

Comment:
The FIXME is wrong for this branch. Currently this branch is dead code I think since your `as_unix_host_fd` always return `Some`, but with @oli-obk's suggestion they could return `None` for FDs which don't have a corresponding host FD.